src/xrpld/rpc/handlers/DepositAuthorized.cpp

```diff
@@ -88,23 +90,108 @@ doDepositAuthorized(RPC::JsonContext& context)
+            return RPC::make_error(
+                rpcINVALID_PARAMS,
+                RPC::expected_field_message(
+                    jss::credentials, "array too long"));
+        }
+
+        for (auto const& jo : creds)
+        {
+            if (!jo.isString())
+            {
+                return RPC::make_error(
+                    rpcINVALID_PARAMS,
+                    RPC::expected_field_message(
+                        jss::credentials, "an array of CredentialID(hash256)"));
+            }
+
+            uint256 credH;
+            auto const credS = jo.asString();
+            if (!credH.parseHex(credS))
+            {
+                return RPC::make_error(
+                    rpcINVALID_PARAMS,
+                    RPC::expected_field_message(
+                        jss::credentials, "an array of CredentialID(hash256)"));
+            }
+
+            std::shared_ptr<SLE const> sleCred =
+                ledger->read(keylet::credential(credH));
+            if (!sleCred || (sleCred->getType() != ltCREDENTIAL) ||
+                !(sleCred->getFlags() & lsfAccepted))
+            {
+                RPC::inject_error(
+                    rpcBAD_CREDENTIALS, "credentials aren't accepted", result);
+                return result;
+            }
+
+            if (credentials::checkExpired(
+                    sleCred, ledger->info().parentCloseTime))
+            {
+                RPC::inject_error(
+                    rpcBAD_CREDENTIALS, "credentials are expired", result);
+                return result;
+            }
+
+            if (reqAuth)
+            {
+                auto credential = STObject::makeInnerObject(sfCredential);
+                credential.setAccountID(sfIssuer, (*sleCred)[sfIssuer]);
+                credential.setFieldVL(
+                    sfCredentialType, (*sleCred)[sfCredentialType]);
+                authCreds.push_back(std::move(credential));
+            }
         }
     }
+
+    // If the two accounts are the same OR if that flag is
+    // not set, then the deposit should be fine.
+    bool depositAuthorized = true;
+
+    if (reqAuth)
+    {
+        if (credentialsPresent)
+        {
+            auto const sorted = credentials::makeSorted(authCreds);
+            if (sorted.empty())
+            {
+                RPC::inject_error(
+                    rpcBAD_CREDENTIALS, "duplicates in crednetials", result);
+                return result;
+            }
+
+            depositAuthorized =
+                ledger->exists(keylet::depositPreauth(dstAcct, sorted));
+        }
+        else
+            depositAuthorized =
+                ledger->exists(keylet::depositPreauth(dstAcct, srcAcct));
+    }
+
     result[jss::source_account] = params[jss::source_account].asString();
     result[jss::destination_account] =
         params[jss::destination_account].asString();
+    if (credentialsPresent)
+        result[jss::credentials] = params[jss::credentials];
```

Comment:
Specification doesn't include `credentials` in the response.

Reply:
@mvadari  please update the specification